conans/client/configure_environment.py

```diff
@@ -1,22 +1,51 @@
 from conans.model.settings import Settings
 import copy
+from conans.client.generators.virtualenv import get_setenv_variables_commands
+from conans.model.env_info import DepsEnvInfo
 
 
 class ConfigureEnvironment(object):
 
-    def __init__(self, deps_cpp_info, settings):
+    def __init__(self, *args):
+        if len(args) == 2:
+            deps_cpp_info = args[0]
+            deps_env_info = DepsEnvInfo()
+            settings = args[1]
+        elif len(args) == 1:  # conanfile (new interface)
+            self.conanfile = args[0]
+            deps_cpp_info = self.conanfile.deps_cpp_info
+            deps_env_info = self.conanfile.deps_env_info
+            settings = self.conanfile.settings
+
         assert isinstance(settings, Settings)
+
         self._settings = settings
         self._deps_cpp_info = deps_cpp_info
-        self.compiler = getattr(self._settings, "compiler", None)
-        self.arch = getattr(self._settings, "arch", None)
-        self.os = getattr(self._settings, "os", None)
-        self.build_type = getattr(self._settings, "build_type", None)
-        self.libcxx = None
+        self._deps_env_info = deps_env_info
+        try:
+            self.compiler = str(self._settings.compiler)
+        except:
+            self.compiler = None
+
+        try:
```

Comment:
But how is it possible that it is not defined? You have an assert a few lines above